src/bundle/Controller/ContentViewController.php

```diff
@@ -59,6 +59,29 @@ public function detailsTabAction(ContentView $view)
         return $view;
     }
 
+    public function versionsTabAction(ContentView $view)
+    {
+        $contentInfo = $view->getContent()->getVersionInfo()->getContentInfo();
+        $contentService = $this->getRepository()->getContentService();
+        $versions = $contentService->loadVersions($contentInfo);
+
+        $versionFilter = $this->container->get('ezsystems.platform_ui.hybrid.filter.version_filter');
```

Comment:
I prefer services to be injected in too. We could make the controller a service that way it can be injected in?
I assumed getting things directly from the container was the convention as that's how it's done elsewhere: https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/MVC/Symfony/Controller/Controller.php#L98